pvlib/ivtools.py

```diff
@@ -262,6 +264,177 @@ def fit_sde_sandia(voltage, current, v_oc=None, i_sc=None, v_mp_i_mp=None,
                                      v_oc)
 
 
+def fit_sdm_desoto(celltype, v_mp, i_mp, v_oc, i_sc, alpha_sc, beta_voc,
+                   cells_in_series, temp_ref=25, irrad_ref=1000):
+    """
+    Calculates the five parameters for the single diode equation using
+    the De Soto et al. procedure described in [1]. This procedure has the
+    advantage of using common specifications given by manufacturers in the
+    datasheets of PV modules.
+    The six values returned by this function can be used by
+    pvsystem.calcparams_desoto to calculate the values at different
+    irradiance and cell temperature.
+
+    Parameters
+    ----------
+    celltype: str, case insensitive
+        Value is one of 'monosi', 'multisi', 'polysi' or'gaas'.
```

Comment:
Ok, then using celltype allowed to be consistent with fit_sdm_cec_sam(). So we should maybe change it in fit_sdm_cec_sam() as well.
Using eV more than J make it inconsistent with scipy.constants (#483), is it still ok ?

Reply:
`scipy.constants` has the value in eV/K, but have to get it using a key rather than as `scipy.constants.Boltzmann`. The statement is in a comment below. So I think it's OK for #483 .

The SAM function requires celltype as an input, and SAM has an internal lookup table for Egref etc.  I'd prefer to drop celltype and pass Egref but that requires a change to SAM.  Your point about consistent signatures is well-taken; feel free to adjust the argument order for fit_sdm_cec_sam() to match, as best we can.